include/exadg/incompressible_navier_stokes/user_interface/parameters.cpp

```diff
@@ -536,9 +548,96 @@ Parameters::check(dealii::ConditionalOStream const & pcout) const
+                  "Variable viscosity parameters stored for cell_based_face_loops == false only."));
+
+    if(temporal_discretization == TemporalDiscretization::BDFDualSplittingScheme)
+    {
+      AssertThrow(treatment_of_nonlinear_viscosity != TreatmentOfNonlinearViscosity::Implicit,
+                  dealii::ExcMessage(
+                    "An implicit-in-time treatment of the nonlinear diffusive term is not possible "
+                    "in combination with the dual splitting scheme."));
+    }
+
+    // check assumptions of rheological models and enforce user to set the parameters
+    // accordingly in the generalized Carreau-Yasuda model
+    if(generalized_newtonian_model == GeneralizedNewtonianModel::Carreau ||
+       generalized_newtonian_model == GeneralizedNewtonianModel::Cross ||
+       generalized_newtonian_model == GeneralizedNewtonianModel::SimplifiedCross)
+    {
+      AssertThrow(std::abs(generalized_newtonian_kappa - 1.0) < 1e-20,
+                  dealii::ExcMessage("generalized_newtonian_kappa == 1 required for "
+                                     "this GeneralizedNewtonianModel."));
+    }
+
+    if(generalized_newtonian_model == GeneralizedNewtonianModel::Carreau)
+    {
+      AssertThrow(std::abs(generalized_newtonian_a - 2.0) < 1e-20,
+                  dealii::ExcMessage("generalized_newtonian_a == 2 required for"
+                                     "GeneralizedNewtonianModel::Carreau."));
+    }
+
+    if(generalized_newtonian_model == GeneralizedNewtonianModel::Cross)
+    {
+      AssertThrow(std::abs(generalized_newtonian_n - 1.0 + generalized_newtonian_a) < 1e-20,
+                  dealii::ExcMessage("generalized_newtonian_n - 1 == generalized_newtonian_a "
+                                     "required for GeneralizedNewtonianModel::Cross."));
+    }
+
+    if(generalized_newtonian_model == GeneralizedNewtonianModel::SimplifiedCross)
+    {
+      AssertThrow(std::abs(generalized_newtonian_a - 1.0) < 1e-20,
+                  dealii::ExcMessage("generalized_newtonian_a == 1 "
+                                     "required for GeneralizedNewtonianModel::SimplifiedCross."));
+      AssertThrow(std::abs(generalized_newtonian_n) < 1e-20,
+                  dealii::ExcMessage("generalized_newtonian_n == 0 "
+                                     "required for GeneralizedNewtonianModel::SimplifiedCross."));
+    }
+
+    if(generalized_newtonian_model == GeneralizedNewtonianModel::PowerLaw)
+    {
+      AssertThrow(std::abs(viscosity) < 1e-20,
+                  dealii::ExcMessage(
+                    "viscosity (= generalized_newtonian_kinematic_viscosity_lower_limit) == 0 "
+                    "required for GeneralizedNewtonianModel::PowerLaw."));
+      AssertThrow(std::abs(generalized_newtonian_kappa) < 1e-20,
+                  dealii::ExcMessage("generalized_newtonian_kappa == 0 required for "
+                                     "required for GeneralizedNewtonianModel::PowerLaw."));
+    }
```

Comment:
I think these checks should not appear in `IncNS::Parameters`. We should follow a more modular design with certain classes caring about consistency of parameters for a particular functionality. This means that we should introduce a separate data struct with all parameters for generalized Newtonian, similarly for the already existing turbulence model stuff, etc.